lib/elixir/lib/kernel.ex

```diff
@@ -6645,7 +6645,15 @@ defmodule Kernel do
   defp compile_regex(binary_or_tuple, options) do
     bin_opts = :binary.list_to_bin(options)
 
-    case is_binary(binary_or_tuple) do
+    # TODO: Remove the second member when we require Erlang/OTP 28+
+    # We defer to a runtime compilation if using the :export flag on older OTP version
+    # that don't support it.
```

Comment:
I think for earlier versions, we should always consider as if export is given, given they can be transmited across nodes and written to config files? So earlier versions don't need to check for it?

Also, why do we want to skip compilation if the "E" flag is given? We can still precompile it, right?

Reply:
The thing is that it would make the compilation fail because of the `:export` flag: https://github.com/sabiwara/elixir/actions/runs/19327180274/job/55281069375

<img width="972" height="348" alt="Screenshot 2025-11-13 at 20 36 10" src="https://github.com/user-attachments/assets/181a529e-b99f-476a-a6fe-49b2637d2cc1" />

What we could do is make `E` a non-op on older versions, e.g. just filter out the flag. `~r/foo/E == ~r/foo/`.
This way, code written with `E` would be compatible on both older versions and newer versions.
It might be a bit confusing, but we could document it and it is only a transitional state of things until we drop OTP27-, so LGTM.

Reply:
OK I think I got it: [ddfc080](https://github.com/elixir-lang/elixir/pull/14907/commits/ddfc0800c329a3b9e23e6762a4e77edba116e85f)